File: leetcode/1716. Calculate Money in Leetcode Bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        sum_ = 0 
        nthMonday = 1
        
        for w in range(n//7, 0, -1):
            sum_ += (2 * nthMonday + 6)  * 7 // 2
            nthMonday += 1
        
        sum_ += (2 * nthMonday + (n % 7 - 1)) * (n % 7) // 2
        return sum_

File: leetcode/1716. Calculate Money in Leetcode Bank.java
class Solution {
    public int totalMoney(int n) {
        int sum = 0, nthMonday = 1;
        
        for(int w = n / 7 ; w > 0 ; --w){
            sum += (2 * nthMonday + 6)  * 7 / 2;
            ++nthMonday;
        }
        
        sum += (2 * nthMonday + (n % 7 - 1)) * (n % 7) / 2;
        
        return sum;
    }
